src/uct/ib/rc/base/rc_iface.c

```diff
@@ -694,14 +709,17 @@ static UCS_CLASS_CLEANUP_FUNC(uct_rc_iface_t)
         ucs_free(self->eps[i]);
     }
 
+    ucs_spin_lock(&self->eps_lock);
```

Comment:
probably not needed in dtor

Reply:
in case if `ucs_list_is_empty(&self->ep_list) == true` - yes, also 
 - there is `ucs_arbiter_cleanup` below, can it invoke some user callback or internal ep_dtor for example?
 - in UCP we have async thread which, in theory, can handle some event this time.

so, in good flow this is almost for free on cleanup, IMHO better to have it

Reply:
seems we are not waiting for anything before doing ucs_spinlock_destroy() below. so if another thread can access the lock, it's broken anyway
we could move the ep_list warning after arbiter cleanup (though it only asserts the arbiter is empty)

Reply:
up to you, will remove